python/cudf/cudf/_lib/column.pyx

```diff
@@ -106,8 +121,36 @@ cdef class Column:
             start = self.offset * self.dtype.itemsize
             end = start + self.size * self.dtype.itemsize
             self._data = self.base_data[start:end]
+            # special case of string column
+            if self.dtype == cudf.api.types.dtype("object"):
+                self._data = self.base_data
         return self._data
 
+    @property
+    def chars_data(self):
+        # special case of string column
+        # if dtype == np.dtype("object"):
+        # this is true for STRING, STRUCT. how to ensure it's string only?
+        if self.dtype != cudf.api.types.dtype("object"):
+            raise TypeError(
+                "Expected a Strings column, "
+                f"got {self.dtype}"
+            )
+        if self.base_data is None:
+            return None
+        cv = self.view()
+        if cv.num_children() == 0:
+            return self.base_data
+        # get the size from offset child column (device to host copy)
+        offsets_column_index = 0
+        offset_child_column = cv.child(offsets_column_index)
+        start = 0
+        end = self.base_data.size
+        if offset_child_column.size() != 0:
+            start = get_element(offset_child_column, cv.offset()).value
+            end = get_element(offset_child_column, cv.offset()+cv.size()).value
+        return self.base_data[start:end]
```

Comment:
@galipremsagar I have to uniquely identify string columns (from struct and other nested columns). How do I do that in Cython column.pyx ?

Reply:
I'm surprised we need to implement this in Cython. This kind of logic for getting the chars of a slice must live _somewhere_ in libcudf, right?

Reply:
Cython column is formed using rmm.DeviceBuffer. So, we don't have column_view. 
Also, there is no existing function for this in string column in libcudf. Should we move this logic to libcudf?

Reply:
What would be moved to `libcudf`? 
The purpose of this PR is to remove the column wrapper for `chars` from libcudf.

Reply:
`the logic for getting the chars of a slice`

Reply:
How will the `chars of a slice` be returned?

Reply:
A buffer?

Reply:
Ok, thanks. Just checking.

Reply:
Here is the only place I can think of that libcudf does this internally.
https://github.com/rapidsai/cudf/pull/14202/files#diff-3ab40a1842ddd5ecded7155994a65580ca0bcfad07bbddc2bf17a59df78394eaR71-R72

Reply:
So I think this could be done by exposing the existing `cudf::strings::detail::copy_slice` API as a public API (`cudf::strings::copy_slice`) and then just retrieve the data-buffer from the returned column.

Reply:
used same logic as `def children(self)`. performs deep copy.